src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs

```diff
@@ -1075,16 +1075,16 @@ internal PseudoBindingInfo DoPseudoParameterBinding(CommandAst command, Type pip
         /// <param name="executionContext">ExecutionContext</param>
         private void SetTemporaryDefaultHost(ExecutionContext executionContext)
         {
-            if (executionContext.EngineHostInterface.IsHostRefSet)
+            if (_restoreHost == null && executionContext.EngineHostInterface.IsHostRefSet)
```

Comment:
I don't see the problem with the current code, can you please explain why the change in this method is needed?

Reply:
Please revert.

Reply:
https://github.com/kwkam/PowerShell/blob/8f98f36f533d4f94679828b5f03517d72e140a2d/src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs#L968-L989

Reply:
By looking at the code where this method is used, `RestoreHost` is called in the `finally` block. And they are manipulating the `executioncontext` from the thread local storage.

It looks to me the variable won't be overwritten as long as you are not passing the same `PseudoParameterBinder` instance to multiple pipeline threads and calling methods on them at the same time.

A `PseudoParameterBinder` instance is not supposed to be shared between pipeline threads, so I don't think we need to change it here. Please revert this change.

https://github.com/kwkam/PowerShell/blob/8f98f36f533d4f94679828b5f03517d72e140a2d/src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs#L968-L989